### PR TITLE
Ready for review: transport and reactive transport

### PR DESCRIPTION
**Pull request by WilkAndy**
ModelGeochemicalDatabase records swap matrix in order to translate bulk compositions back to original basis.  This swap matrix is updated every time a swap is performed by the GeochemistrySpeciesSwapper.  GeochemicalSystem has a method for getting the bulk in the original basis.

ModelGeochemicalDatabase has boolean vectors (basis_species_transported, eqm_species_transported and kin_species_transported) that define whether the species is involved in the transport side of reactive-transport.  GeochemistrySpeciesSwapper swaps these appropriate upon swapping

GeochemicalSystem has methods for computing the bulk-composition mole number of transporting species, both in the current basis and the original basis, based on the current values of molalities

Refs #15693

### CODE REVIEW

**Comment by loganharbour** [comment]
Use the draft option :-)

**Comment by WilkAndy** [comment]
Hey @cpgr,

You might be interested in the latest commit.  Adaptive timestepping is implemented in the SpatialReactor which might help with convergence.  You still have to set your nodal_void_volume correctly, otherwise you might (will) end up subtracting a species from a node that doesn't have that species.

a

**Comment by WilkAndy** [comment]
Hey @cpgr , you can see where your figures fit in at the bottom of

https://mooseframework.inl.gov/docs/PRs/15698/site/modules/geochemistry/tests_and_examples/geotes_weber_tensleep.html

a

**Comment by cpgr** [comment]
Great - do you want me to push them here when I've finished or just send them to you?

**Comment by WilkAndy** [comment]
I think just email them  as i've got a few more amendments to make too.

a

________________________________

**Comment by lindsayad** [comment]
Looks like you have a module failure

**Comment by WilkAndy** [comment]
> Looks like you have a module failure

Yea, i'm trying to get to the bottom of that.  Only happens for a job with 2 processors running 2 threads on Minimum clang version.   So difficult to me to get to the root problem.  I'm going to mail moose-users.

**Comment by WilkAndy** [comment]
Reminding you about this monumental PR, @loganharbour

**Comment by loganharbour** [comment]
Looks good to me @WilkAndy.

Will get it in once we can get stuff passing again.

**Comment by loganharbour** [comment]
Also: if you keep adding all of this media content (images), I would prefer that you start putting them into large_media.

**Comment by WilkAndy** [comment]
At the moment i don't understand why the spatial_4_threads on Minimum clang version is failing.  I'm still investigating as i can't reproduce the failure on any of my systems.

**Comment by loganharbour** [comment]
> Warning: This PR changes repo size by 5.40 MiB.

I actually missed this further up. Is it possible for you to move the images into large_media?

**Comment by WilkAndy** [comment]
Hi @loganharbour .  So, what i've done is to

1. rebase against next.  This meant a whole lot of trouble because lots of the unit tests failed because of the changes to json in the rest of moose, sigh
2. squashed everything into one giant commit - this just made the next step easier
3. removed the png files because i've added them to large_media.

If everything passes i'll squash (2) and (3) together to properly remove the png files, but i'm using CIVET to check everything, partly because it's more thorough than my testing, and partly because it's a lot faster.

File: large_media [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 77f85490982aa6357b5e9b062ee28694b8dc385d
+Subproject commit e41f7c8f6e0e77b620dbe4a2b24711b89f52599e
```
[the remark this answers is not in the document]
Comment (WilkAndy) [reply large_media:1]: Yep, that's deliberate

**Comment by WilkAndy** [comment]
Hey @loganharbour , finally all the tests have passed.  I added the png files to large_media.   I really hope you can merge this - i spent most of the day mucking around on things that had regressed through changes in the framework, which was a bit disheartening!

**Review by loganharbour — APPROVED** [review]
Good to go once next is green again!

**Comment by WilkAndy** [comment]
Sounds good, do i have to do anything further, @loganharbour ?

**Comment by lindsayad** [comment]
https://civet.inl.gov/job/617745/

**Comment by lindsayad** [comment]
I thought that you had decided to skip that test

**Pull request merged**